packages/lib/src/components/internal/Button/CopyButton.tsx

```diff
@@ -0,0 +1,60 @@
+import { h } from 'preact';
+import { useCallback } from 'preact/hooks';
+import Button from './Button';
+import { ButtonProps } from './types';
+import { useCoreContext } from '../../../core/Context/CoreProvider';
+import copyToClipboard from '../../../utils/clipboard';
+import { PREFIX } from '../Icon/constants';
+import useImage from '../../../core/Context/useImage';
+
+export interface CopyButtonProps extends Omit<ButtonProps, 'onClickCompletedLabel'> {
+    /**
+     * String that will get copied to the clipboard
+     */
+    text: string;
+    copiedLabel?: string;
+    onClick?: (e: h.JSX.TargetedMouseEvent<HTMLButtonElement>) => void;
+}
+
+const CopyButton = (props: CopyButtonProps) => {
+    const { i18n } = useCoreContext();
+    const getImage = useImage();
+
+    const onClick = useCallback(
+        (e, { complete }) => {
+            void copyToClipboard(props.text);
+            complete();
+            props.onClick?.(e);
+        },
+        [props.text, props.onClick]
+    );
+
+    // todo: we should remove it from the whole sdk: onKeyPress is deprecated.
+    const onKeyPress = useCallback((event: KeyboardEvent) => {
+        if (event.key === 'Enter' || event.code === 'Enter' || event.key === ' ' || event.code === 'Space') {
```

Comment:
Might make sense to move this to a reusuable function since we use this logic in quite some places now e.g `isFocusedElementKeyPress(event)`